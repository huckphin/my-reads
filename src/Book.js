import React, { Component } from 'react';
import BookAuthor from './BookAuthor';
import BookTitle from './BookTitle';
import BookShelfChanger from './BookShelfChanger';
import PropType from 'prop-types';

class Book extends Component {
    static propTypes = {
        authors: PropType.array.isRequired,
        imageLinks: PropType.object.isRequired,
        title: PropType.string.isRequired
    }

    render() {
        const { authors, imageLinks, title } = this.props;

        return (
            <div className="book">
                <div className="book-top">
                    <div
                        className="book-cover"
                        style={{
                            height: 188,
                            width: 192,
                            backgroundImage: `url(${imageLinks.smallThumbnail})` }}>
                    </div>
                    <BookShelfChanger />
                </div>
                <BookAuthor authors={ authors } />
                <BookTitle title={ title } />
            </div>
        );
    };
}

export default Book