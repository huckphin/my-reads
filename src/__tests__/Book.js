import React from 'react';
import Book from '../Book';
import renderer from 'react-test-renderer';

test('will render a Book', () => {
  const onUpdateBook = jest.fn();
  const book = {"title": "Fight Club"};
  const id = "book-id";

  const component = renderer.create(
    <Book
          book={ book }
          key={ id }
          onUpdateBook={ onUpdateBook }
          imageLinks={{
            smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
            thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
          }}
      />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('will render a Book with no imageLinks', () => {
  const onUpdateBook = jest.fn();
  const book = {"title": "Fight Club"};
  const id = "book-id";

  const component = renderer.create(
      <Book
          book={ book }
          key={ id }
          onUpdateBook={ onUpdateBook }
      />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
