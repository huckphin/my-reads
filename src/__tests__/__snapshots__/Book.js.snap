// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render a Book 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(null)",
          "height": 188,
          "width": 192,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select
        defaultValue={undefined}
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="move-to"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-authors"
  >
    Unknown author
  </div>
  <div
    className="book-title"
  >
    Fight Club
  </div>
</div>
`;

exports[`will render a Book with no imageLinks 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(null)",
          "height": 188,
          "width": 192,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select
        defaultValue={undefined}
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="move-to"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-authors"
  >
    Unknown author
  </div>
  <div
    className="book-title"
  >
    Fight Club
  </div>
</div>
`;
